pkg/operator/controller/ingress/load_balancer_service_test.go

```diff
@@ -132,32 +212,94 @@ func TestDesiredLoadBalancerService(t *testing.T) {
+					t.Errorf("annotation check for test %q failed; unexpected annotation %s: %s", tc.description, iksLBScopeAnnotation, iksLBScopePublic)
+				}
+			} else {
+				if err := checkServiceHasAnnotation(svc, iksLBScopeAnnotation, true, iksLBScopePublic); err != nil {
+					t.Errorf("annotation check for test %q failed: %v", tc.description, err)
+				}
+				// The ibm private annotation value should not exist for an external LB.
+				if err := checkServiceHasAnnotation(svc, iksLBScopeAnnotation, true, iksLBScopePrivate); err == nil {
+					t.Errorf("annotation check for test %q failed; unexpected annotation %s: %s", tc.description, iksLBScopeAnnotation, iksLBScopePrivate)
+				}
 			}
-			switch {
-			case tc.lbStrategy.ProviderParameters == nil || tc.lbStrategy.ProviderParameters.AWS.Type == operatorv1.AWSClassicLoadBalancer:
-				if err := checkServiceHasAnnotation(svc, awsLBProxyProtocolAnnotation, true, "*"); err != nil {
+		case configv1.AzurePlatformType:
+			if isInternal {
+				if err := checkServiceHasAnnotation(svc, azureInternalLBAnnotation, true, "true"); err != nil {
 					t.Errorf("annotation check for test %q failed: %v", tc.description, err)
 				}
-			case tc.lbStrategy.ProviderParameters.AWS.Type == operatorv1.AWSNetworkLoadBalancer:
-				if err := checkServiceHasAnnotation(svc, AWSLBTypeAnnotation, true, AWSNLBAnnotation); err != nil {
+			} else {
+				// The azure private annotation should not exist for external LBs.
+				if err := checkServiceHasAnnotation(svc, azureInternalLBAnnotation, false, ""); err == nil {
+					t.Errorf("annotation check for test %q failed; unexpected annotation %s", tc.description, azureInternalLBAnnotation)
+				}
+			}
+		case configv1.GCPPlatformType:
+			if isInternal {
+				if err := checkServiceHasAnnotation(svc, gcpLBTypeAnnotation, true, "Internal"); err != nil {
 					t.Errorf("annotation check for test %q failed: %v", tc.description, err)
 				}
-			case tc.lbStrategy.Scope == operatorv1.InternalLoadBalancer:
-				if err := checkServiceHasAnnotation(svc, AWSLBTypeAnnotation, true, "0.0.0.0/0"); err != nil {
+			} else {
+				// The internal gcp annotation should not exist for an external LB.
+				if err := checkServiceHasAnnotation(svc, gcpLBTypeAnnotation, false, ""); err == nil {
+					t.Errorf("annotation check for test %q failed; unexpected annotation %s", tc.description, gcpLBTypeAnnotation)
+				}
+			}
+		case configv1.OpenStackPlatformType:
+			if isInternal {
+				if err := checkServiceHasAnnotation(svc, openstackInternalLBAnnotation, true, "true"); err != nil {
 					t.Errorf("annotation check for test %q failed: %v", tc.description, err)
 				}
+			} else {
+				// The internal openstack annotation should not exist for an external LB.
+				if err := checkServiceHasAnnotation(svc, openstackInternalLBAnnotation, false, ""); err == nil {
+					t.Errorf("annotation check for test %q failed; unexpected annotation %s", tc.description, openstackInternalLBAnnotation)
+				}
 			}
 		}
```

Comment:
Instead of having detailed logic for each platform to check for the expected annotations, could you put each test case's expected annotations in a map in the test case and compare the actual annotations map with the expected map from the test case?